src/Transition/CreateNewIDFUsingRulesV9_5_0.f90

```diff
@@ -399,6 +399,23 @@ SUBROUTINE CreateNewIDFUsingRules(EndOfFile,DiffOnly,InLfn,AskForInput,InputFile
                                                 'The air boundary will be modeled using the GroupedZones method.')
                  END IF
                  
+             CASE('CONSTRUCTION:INTERNALSOURCE')
+                 ! Do not write the old object
+                 Written=.true.
+                 
+                CALL GetNewObjectDefInIDD('ConstructionProperty:InternalHeatSource',NwNumArgs,NwAorN,NwReqFld,NwObjMinFlds,NwFldNames,NwFldDefaults,NwFldUnits)
+                OutArgs(1) = TRIM(InArgs(1)) // ' Heat Source'
+                OutArgs(2) = InArgs(1)
+                OutArgs(3:7) =  InArgs(2:6)
+                CALL WriteOutIDFLines(DifLfn,'ConstructionProperty:InternalHeatSource',NwNumArgs,OutArgs,NwFldNames,NwFldUnits)
+                
+                CALL GetNewObjectDefInIDD('Construction',NwNumArgs,NwAorN,NwReqFld,NwObjMinFlds,NwFldNames,NwFldDefaults,NwFldUnits)
+                ! Construction object has five fewer fields than the incoming Construction:InternalSource object
+                NwNumArgs = CurArgs - 5
+                OutArgs(1) = InArgs(1)
+                OutArgs(2:NwNumArgs) = InArgs(7:CurArgs)
+                CALL WriteOutIDFLines(DifLfn,'Construction',NwNumArgs,OutArgs,NwFldNames,NwFldUnits)
```

Comment:
I was about to comment on how you also wrote out the object accidentally but I see you leveraged the `written` variable above to avoid it.  Looks good.